arch/risc-v/src/common/espressif/esp_wifi_utils.c

```diff
@@ -0,0 +1,643 @@
+      if (iwr->u.data.flags & IW_SCAN_THIS_ESSID &&
+          req->essid_len < sizeof(target_ssid))
+        {
+          /* Scan specific ESSID */
+
+          config->show_hidden = true;
+          config->bssid = NULL;
+          memcpy(&target_ssid[0], req->essid, req->essid_len);
+          config->ssid = &target_ssid[0];
+          config->ssid[req->essid_len] = '\0';
+        }
+
+      if (iwr->u.data.flags & IW_SCAN_THIS_FREQ &&
+          req->num_channels > 0)
+        {
+          /* Scan specific channels */
+
+          DEBUGASSERT(req->num_channels <= CHANNEL_MAX_NUM);
+          g_channel_num = req->num_channels;
+          if (req->num_channels == 1)
+            {
+              config->channel = req->channel_list[0].m;
+            }
+          else
+            {
+              for (i = 0; i < req->num_channels; i++)
+                {
+                  g_channel_list[i] = req->channel_list[i].m;
+                }
+            }
+        }
+
+      memset(target_mac, 0xff, MAC_LEN);
+      if (memcmp(req->bssid.sa_data, target_mac, MAC_LEN) != 0)
+        {
+          /* Scan specific bssid */
+
+          memcpy(target_mac, req->bssid.sa_data, MAC_LEN);
+          config->bssid = &target_mac[0];
+        }
+    }
+  else
+    {
+      /* Default scan parameters */
+
+      wlinfo("INFO: Use default scan parameters\n");
+      config->scan_type = WIFI_SCAN_TYPE_ACTIVE; /* Active scan */
+    }
+
+  esp_wifi_start();
+  ret = esp_wifi_scan_start(config, false);
+  if (ret != OK)
+    {
+      wlerr("ERROR: Scan error, ret: %d\n", ret);
+    }
+  else
+    {
+      /* Allocate buffer to store scan result */
+
+      if (priv->scan_result == NULL)
+        {
+          priv->scan_result = kmm_malloc(WIFI_SCAN_RESULT_SIZE);
+          if (priv->scan_result == NULL)
+            {
+              wlerr("ERROR: Cannot allocate result buffer\n");
+              ret = -ENOMEM;
+            }
+          else
+            {
+              memset(priv->scan_result, 0x0, WIFI_SCAN_RESULT_SIZE);
+            }
+        }
+    }
+
+  if (config)
+    {
+      kmm_free(config);
+      config = NULL;
+      wlinfo("INFO: start scan\n");
+    }
+
+  g_scan_priv.scan_status = ESP_SCAN_RUN;
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: esp_wifi_get_scan_results
+ *
+ * Description:
+ *   Get scan result
+ *
+ * Input Parameters:
+ *   iwr - The argument of the ioctl cmd
+ *
+ * Returned Value:
+ *   OK on success (positive non-zero values are cmd-specific)
+ *   Negated errno returned on failure.
+ *
+ ****************************************************************************/
+
+int esp_wifi_get_scan_results(struct iwreq *iwr)
+{
+  int ret = OK;
+  static bool scan_block = false;
+  struct wifi_scan_result *priv = &g_scan_priv;
+
+  if (g_scan_priv.scan_status == ESP_SCAN_RUN)
+    {
+      irqstate_t irqstate = enter_critical_section();
+      if (scan_block == false)
```

Comment:
```suggestion
      if (!scan_block)
```

Reply:
Hmm, that's something that I would prefer to keep as it is for the sake of consistency. Particularly, I couldn't find any standard on NuttX and I even already have received suggestions to make it explicit beforehand. In this case, other similar sources (like `arch/xtensa/src/esp32s3/esp32s3_wifi_utils.c`) use it explicitly, so I prefer to keep it as it is instead of modifying it in the other sources if you don't mind. What do you think about it?